hack/lib/util/environment.sh

```diff
@@ -231,6 +231,10 @@ readonly -f os::util::environment::setup_etcd_vars
 #  - export MASTER_CONFIG_DIR
 #  - export NODE_CONFIG_DIR
 function os::util::environment::setup_server_vars() {
+    # turn on cache mutation detector every time we start a server
+    KUBE_CACHE_MUTATION_DETECTOR="${KUBE_CACHE_MUTATION_DETECTOR:-true}"
```

Comment:
@kargakis I don't speak bash very well, this does what I think it does, right?  Defaults to true if its not set, but allows setting `KUBE_CACHE_MUTATION_DETECTOR=false hack/test-cmd.sh`